src/Features/CSharp/Portable/Completion/KeywordRecommenders/EventKeywordRecommender.cs

```diff
@@ -25,6 +25,23 @@ internal class EventKeywordRecommender : AbstractSyntacticSingleKeywordRecommend
                 SyntaxKind.UnsafeKeyword
             };
 
+        private static readonly ISet<SyntaxKind> s_validStructModifiers = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
+            {
+                SyntaxKind.NewKeyword,
+                SyntaxKind.PublicKeyword,
+                SyntaxKind.ProtectedKeyword,
+                SyntaxKind.InternalKeyword,
+                SyntaxKind.PrivateKeyword,
+                SyntaxKind.StaticKeyword,
+                SyntaxKind.VirtualKeyword,
+                SyntaxKind.SealedKeyword,
+                SyntaxKind.OverrideKeyword,
+                SyntaxKind.AbstractKeyword,
+                SyntaxKind.ExternKeyword,
+                SyntaxKind.UnsafeKeyword,
+                SyntaxKind.ReadOnlyKeyword,
```

Comment:
Is this only different in that it also has ReadOnlyKeyword? I'm wondering if we're best off just having this be something like:

    private static readonly ISet<SyntaxKind> s_validStructModifiers = new HashSet<SyntaxKind>(s_validClassModifiers.Concat(SyntaxKind.ReadOnlyKeyword))

Since otherwise we might make some mistakes here where a new modifier generally available to events we now have to remember to update twice?

Reply:
I wasn't sure, and so I took the least clever option. But I'm happy to make the change.

Reply:
early on we kept this all very segmented in case we neede to make very targetted fixes without impacting other keywords.  However, with the amount of testing we have, and how well baked this is, i'm fine with writing clearer code that has less duplication.   So @jasonmalinowski 's suggestion is 100% ok for me.

Reply:
@CyrusNajmabadi If this is the preferred style then I'm more than happy to take as is. Up to you since you know this better than I.

Reply:
At the time it was... but i was super paranoid about wanting to be very explicit.  This was based on the pain we had in the native system where there were tons of dependencies between all the keywords, so touching one would break 5 others in 15 random niche cases.  The new system is much clearer and has avoided that problem, and you also did a great job on the VB side, showing how this all could work.  So i'm 100% ok with simplification now that we've shown we're in a good position without the pain we used to have :)

Reply:
+1 for this change.

Reply:
Change made.